actions/setup/action.yml

```diff
@@ -43,6 +43,18 @@ runs:
         check-latest: ${{ inputs.check-latest }}
         cache: false  # cache is handled by separate actions/cache step.
 
+    - name: Generate cache key
+      id: cache-key
+      shell: bash
+      run: |
+        if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+          echo "cache_key=${{ runner.os }}-sage-pr-${{ github.base_ref }}-${{ inputs.cacheKey }}-${{ inputs.go-version }}-${{ hashFiles('**/go.sum') }}" >> $GITHUB_OUTPUT
+          echo "restore_key=${{ runner.os }}-sage-pr-${{ github.base_ref }}-${{ inputs.cacheKey }}-${{ inputs.go-version }}-" >> $GITHUB_OUTPUT
+        else
```

Comment:
else, should that only create if the branch is `master` or `main`? or do want to create for all branches?

Reply:
lets keep it simple for now and keep it as is